source/adapter.screenshare.js

```diff
@@ -175,85 +339,219 @@ AdapterJS.defineMediaSourcePolyfill = function () {
-    // nothing here because edge does not support screensharing
+    // Note: Not overriding getUserMedia() to reject "mediaSource" as to prevent "Invalid calling object" errors.
+    // Nothing here because edge does not support screensharing
     console.warn('Edge does not support screensharing feature in getUserMedia');
 
   } else {
     baseGetUserMedia = window.navigator.getUserMedia;
 
     navigator.getUserMedia = function (constraints, successCb, failureCb) {
-      if (constraints && constraints.video && !!constraints.video.mediaSource) {
-        // would be fine since no methods
+      // Append checks for overrides as these are mandatory
+      // Browsers (not Firefox since they went Promise based) does these checks and they can be quite useful
+      if (!(constraints && typeof constraints === 'object')) {
+        throw new Error('GetUserMedia: (constraints, .., ..) argument required');
+      } else if (typeof successCb !== 'function') {
+        throw new Error('GetUserMedia: (.., successCb, ..) argument required');
+      } else if (typeof failureCb !== 'function') {
+        throw new Error('GetUserMedia: (.., .., failureCb) argument required');
+      }
+
+      if (constraints.video && typeof constraints.video === 'object' && constraints.video.hasOwnProperty('mediaSource')) {
         var updatedConstraints = clone(constraints);
 
-        // wait for plugin to be ready
+        // Wait for plugin to be ready
         AdapterJS.WebRTCPlugin.callWhenPluginReady(function() {
-          // check if screensharing feature is available
-          if (!!AdapterJS.WebRTCPlugin.plugin.HasScreensharingFeature &&
-            !!AdapterJS.WebRTCPlugin.plugin.isScreensharingAvailable) {
-            // set the constraints
+          // Check if screensharing feature is available
+          if (!!AdapterJS.WebRTCPlugin.plugin.HasScreensharingFeature && !!AdapterJS.WebRTCPlugin.plugin.isScreensharingAvailable) {
+            // Do strict checks for the source ID - "screen", "window" or ["screen", "window"]
+            var sourceId = AdapterJS.WebRTCPlugin.plugin.screensharingKey || 'Screensharing';
+
+            if (AdapterJS.WebRTCPlugin.plugin.screensharingKeys) {
+              // Param: ["screen", "window"]
+              if (Array.isArray(updatedConstraints.video.mediaSource) && 
+                updatedConstraints.video.mediaSource.indexOf('screen') > -1 &&
+                updatedConstraints.video.mediaSource.indexOf('window') > -1) {
+                sourceId = AdapterJS.WebRTCPlugin.plugin.screensharingKeys.screenOrWindow;
+                updatedConstraints.video.mediaSource = AdapterJS.WebRTCPlugin.plugin.screensharingKeys.screenOrWindow;
+              // Param: ["screen"] or "screen"
+              } else if ((Array.isArray(updatedConstraints.video.mediaSource) && 
+                updatedConstraints.video.mediaSource.indexOf('screen') > -1) || updatedConstraints.video.mediaSource === 'screen') {
```

Comment:
@miniruwan it's expected. It should return the Array of the index so `> -1` would be correct since `0` greater than `-1` as according to [`Array.indexOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf?v=example).